concordium-sdk/src/main/java/com/concordium/sdk/ClientV2MapperExtensions.java

```diff
@@ -1245,4 +1249,28 @@ static com.concordium.sdk.responses.blockinfo.BlockInfo to(BlockInfo blockInfo)
                 .blockArriveTime(to(to(blockInfo.getArriveTime())))
                 .build();
     }
+
+    static FinalizationData to(BlockFinalizationSummary finalizationSummary) {
+        if (finalizationSummary.hasNone()) {return null;} //There is no finalization data in the block
+        val finalizationData = finalizationSummary.getRecord();
+        val finalizationBlockPointer = to(finalizationData.getBlock()).toString();
+        int finalizationIndex = (int) finalizationData.getIndex().getValue();
+        int finalizationDelay = (int) finalizationData.getDelay().getValue();
```

Comment:
In principle that is a loss of precision since int is 32-bits. It would be ideal to have this UInt64, at least for the index. Delay is very unlikely to overflow.

Reply:
I understand that involves changing the FinalizationData type, but it is the correct thing to do.

Reply:
It would, but ever so slightly. It is good to fix the bugs though.